dags/common/pull_ftp.py

```diff
@@ -21,21 +22,26 @@ def migrate_files(
 ):
     logger.msg("Processing files.", filenames=filenames)
     extracted_filenames = []
-    for _file in filenames:
-        logger.msg("Getting file from SFTP.", file=_file)
-        file_bytes = s_ftp.get_file(_file)
-        if not zipfile.is_zipfile(file_bytes):
-            logger.msg("File is not a zipfile, processing next file.")
-            continue
-        with zipfile.ZipFile(file_bytes) as zip:
-            for zip_filename in zip.namelist():
-                file_prefix = ".".join(_file.split(".")[:-1])
-                file_content = zip.read(zip_filename)
-                s3_filename = os.path.join(file_prefix, zip_filename)
-                repo.save(s3_filename, io.BytesIO(file_content))
+
+    for file_name in filenames:
```

Comment:
maybe archive_name? Because it can be zip or tar

Reply:
just an idea: can't you filter files that you pull from ftp so that you get files with desired extensions only? do they really put everything in one directory on ftp? 🤦‍♀️

Reply:
Sorry, it was wrong what I said. Actually it can be zip or tar, that's it. ZIP for IOP, OUP, Springer, TAR for Elsevier

Reply:
but inside of archive, we have xml, txt files.
I don't think we can call tar_file maybe `archive_file` is better?